acquia_lift.admin.inc

```diff
@@ -287,36 +304,110 @@ function acquia_lift_admin_form_submit($form, &$form_state) {
+      ),
+    ),
+  );
+  if (!acquia_lift_is_targeting_agent($agent_data)) {
+    $legacy_campaigns = variable_get('acquia_lift_legacy_agents', array());
+    if (in_array($agent_data->machine_name, $legacy_campaigns)) {
+      $form['report'] = acquia_lift_report_custom($form_state, $agent_data, FALSE);
+      return $form;
     }
-    foreach($option_set->targeting as $audience_name => $audience) {
-      if (isset($audience['osid'])) {
-        if ($option_set = personalize_option_set_load($audience['osid'])) {
-          $nested_agent = personalize_agent_load($option_set->agent);
-          if (!empty($nested_agent)) {
-            $report[$nested_agent->machine_name] = array(
-              'title' => array(
-                '#markup' => '<h2>' . t('Test report for audience @audience', array('@audience' => $audience_name)) . '</h2>'
-              ),
-              'report' => acquia_lift_report($form, $form_state, $nested_agent, NULL, TRUE)
-            );
-          }
-        }
-      }
+    else {
+      return array();
+    }
+  }
+  $option_set = acquia_lift_get_option_set_for_targeting($agent_data->machine_name);
+
+  if (empty($option_set->targeting)) {
+    return array();
+  }
+  // Display a report for the specified audience.
+  if (isset($form_state['values']['audience_filter'])) {
+    $audience = $form_state['values']['audience_filter'];
+  }
+  else {
+    $audience = isset($_GET['targeting_audience']) ? $_GET['targeting_audience'] : NULL;
+  }
+  // Get audience options.
+  $audience_options = array();
+  foreach($option_set->targeting as $audience_name => $targeting) {
+    if (isset($targeting['osid'])) {
+      $audience_options[$audience_name] = $targeting['label'];
     }
-    return $report;
   }
-  elseif (!$nested && acquia_lift_is_testing_agent($agent_data)) {
-    $legacy = TRUE;
+
+  if (empty($audience) && !isset($option_set->targeting[$audience]['osid'])) {
+    if (empty($audience_options)) {
+      return array(
+        'no_report' => array(
+          '#markup'=> t('This personalization does not contain tests, no reports to show.')
+        )
+      );
+    }
+    $audience = key($audience_options);
+  }
+
+  if ($nested_os = personalize_option_set_load($option_set->targeting[$audience]['osid'])) {
+    $nested_agent = personalize_agent_load($nested_os->agent);
+    if (!empty($nested_agent)) {
+      if (count($audience_options) == 1) {
+        $form['audience_label'] = array(
+          '#markup' => '<h2>' . t('Test for %audience audience', array('%audience' => $option_set->targeting[$audience]['label'])) . '</h2>'
+        );
+        $form['audience_filter'] = array(
+          '#type' => 'value',
+          '#value' => $audience
+        );
+      }
+      else {
+        $form['audience_filter'] = array(
+          '#title' => t('Audience'),
+          '#type' => 'select',
+          '#options' => $audience_options,
+          '#default_value' => $audience,
+          '#attributes' => array(
```

Comment:
Darn - this only adds to the input... it needs to be on the wrapping form element to actually have the desired effect.  Probably not worth the effort at this point for a theme function change just for this.